src/algebra/big_operators/finprod.lean

```diff
@@ -205,6 +205,14 @@ open_locale big_operators
   ∏ᶠ (h : a ∈ s), f a = mul_indicator s f a :=
 by convert finprod_eq_if
 
+@[to_additive] lemma finprod_mem_mul_support (f : α → M) (a : α):
```

Comment:
```suggestion
@[simp, to_additive] lemma finprod_mem_mul_support (f : α → M) (a : α) :
```

Reply:
Annoyingly I can't do this because `mem_mul_support` is tagged as `simp` :-/

Reply:
Perhaps it would be better thhen to change the statement to `∏ᶠ (h : f a ≠ 1), f a = f a` then? Or do we already have that?

Reply:
We do not have that already but I think what we should really do is drop the `simp` on `mem_mul_support`.

I note also that even though I certainly can adjust `finprod_mem_mul_support` as you suggest (at the cost of sprinkling in a few `mem_mul_support` in places) I cannot change the statement of `finprod_along_mul_support` to `∏ᶠ (f a ≠ 1), f a = ∏ᶠ a, f a`.

Reply:
Sure, but you could change it to `∏ᶠ a (H : f a ≠ 1), f a = ∏ᶠ a, f a`. Remember that `∏ᶠ (a ∈ mul_support f), f a` is just shorthand for `∏ᶠ a (H: a ∈ mul_support f), f a`, they are indistinguishable to lean.

Reply:
Sorry my remarks above was a bit of a thinko and isn't even relevent anyway. I think you're probably right about just restating this lemma as `∏ᶠ (h : f a ≠ 1), f a = f a` but I've added https://github.com/leanprover-community/mathlib/pull/7439/commits/db43cf0ad556118f50b87a83afce3624b2591771 as an experiment to see if it builds. I'll sort this out tomorrow.

Reply:
OK I think I've pushed up something sane now.